apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -302,7 +311,8 @@ def _perform_gw_port_operations(self, context, port):
         router_id = port.get('device_id')
         network = context.network.current
         router_info = self.apic_manager.ext_net_dict.get(network['name'])
-
+        tenant_name = (self.tenant_common or
+                       self.name_mapper.tenant(context, network['tenant_id']))
         if router_id and router_info:
```

Comment:
Nope, in apic ML2 driver we don't put shared networks in the common tenant. They still are owned by "tenant_id", and visible through Openstack to anyone, what we do is carefully picking the tenant to use when a port is created in that network (basically, a port owned by tenant X in Openstack could be placed in tenant Y in ACI because he owns the Network).